conda/plugins/__init__.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+import pluggy
+import sys
+
+from typing import Callable, List, NamedTuple, Optional
+
+
+if sys.version_info < (3, 9):
+    from typing import Iterable
+else:
+    from collections.abc import Iterable
+
+
+_hookspec = pluggy.HookspecMarker("conda")
+register = pluggy.HookimplMarker("conda")
+
+
+class CondaSubcommand(NamedTuple):
+    """
+    Conda subcommand entry.
+
+    :param name: Subcommand name (e.g., ``conda my-subcommand-name``).
+    :param summary: Subcommand summary, will be shown in ``conda --help``.
+    :param action: Callable that will be run when the subcommand is invoked.
+    """
+    name: str
+    summary: str
+    action: Callable[
+        [List[str]],  # arguments
+        Optional[int],  # return code
+    ]
+
```

Comment:
```suggestion
from __future__ import annotations

import pluggy
import sys

from typing import Callable, NamedTuple, Optional
from collections.abc import Iterable


_hookspec = pluggy.HookspecMarker("conda")
register = pluggy.HookimplMarker("conda")


class CondaSubcommand(NamedTuple):
    """
    Conda subcommand entry.

    :param name: Subcommand name (e.g., ``conda my-subcommand-name``).
    :param summary: Subcommand summary, will be shown in ``conda --help``.
    :param action: Callable that will be run when the subcommand is invoked.
    """
    name: str
    summary: str
    action: Callable[
        [list[str]],  # arguments
        Optional[int],  # return code
    ]

```

I don't think it's necessary to do a version check before importing `Iterable` from `collections.abc`. That has been available since `3.7` which is the earliest version we support.

Also, let's use the `from __future__ import annotations` statement where possible so we can avoid having to import the builtin types like `List` from the `typing` module.